JSX in depth PL translation (#181)
reactjs/pl.reactjs.org: content/docs/jsx-in-depth.md

  - "docs/jsx-in-depth-ko-KR.html"
---

Zasadniczo, JSX dostarcza lukier składniowy dla funkcji `React.createElement(component, props, ...children)`. Kod JSX:

```js
<MyButton color="blue" shadowSize={2}>

>   Click me

Reviewer: ```suggestion
  Kliknij mnie
```

</MyButton>
```

jest kompilowany do:

```js
React.createElement(
  MyButton,
  {color: 'blue', shadowSize: 2},

>   'Click me'

Reviewer: ```suggestion
  'Kliknij mnie'
```

)
```

> Aby przetestować jak JSX jest zamieniany na JavaScript, możesz wypróbować [kompilator Babel online](babel://jsx-simple-example).

Reviewer: ```suggestion
Aby przetestować, jak JSX jest zamieniany na JavaScript, możesz wypróbować [kompilator Babel online](babel://jsx-simple-example).
```

Reviewer: Racja, ale przecinek musi być! :)

> Aby przetestować jak JSX jest zamieniany na JavaScript, możesz wypróbować [kompilator Babel online](babel://jsx-simple-example).

Reviewer: ```suggestion
Aby przetestować, jak JSX jest zamieniany na JavaScript, możesz wypróbować [kompilator Babel online](babel://jsx-simple-example).
```

## Określanie typu elementu {#specifying-the-react-element-type}

> Pierwsza część tagu JSX określa typ elementu Reactowego.

Reviewer: ```suggestion
Pierwsza część tagu JSX określa typ elementu reactowego.
```

Reviewer: @stepek zbędny ten przecinek.

> Wielka litera w tagu JSX oznacza, że jest on komponentem Reacta. Takie tagi są bezpośrednio kompilowane na referencję do zmiennej, więc jeżeli używasz w swoim kodzie JSX wyrażenia `<Foo />`, to `Foo` musi znajdować się w zakresie.

Reviewer: ```suggestion
Wielka litera w tagu JSX oznacza, że jest on komponentem reactowym. Takie tagi są bezpośrednio kompilowane na referencję do zmiennej, więc jeżeli używasz w swoim kodzie JSX wyrażenia `<Foo />`, to `Foo` musi znajdować się w zakresie.
```

### React musi znajdować się w zakresie {#react-must-be-in-scope}

> Jako, że JSX kompiluje się do wywołania `React.createElement`, biblioteka `React` musi być w zakresie, w którym znajduje się Twój kod JSX.

Reviewer: ```suggestion
Jako że JSX kompilowany jest do wywołania `React.createElement`, biblioteka `React` musi być w zakresie, w którym znajduje się twój kod JSX.
```

> W przykładzie poniżej, oba importy są konieczne nawet jeżeli `React` i `CustomButton` nie mają żadnych bezpośrednich odniesień z kodu JavaScript:

Reviewer: ```suggestion
W przykładzie poniżej oba importy są konieczne, nawet jeżeli `React` i `CustomButton` nie mają żadnych bezpośrednich odniesień z kodu JavaScript:
```

}
```

> Jeżeli nie używasz rozwiązania budującego paczkę z kodem JavaScript i ładujesz Reacta za pomocą tagu `<script>`, jest on dostępny w globalnym zakresie pod etykietą `React`.

Reviewer: ```suggestion
Jeżeli nie używasz rozwiązania budującego paczkę z kodem JavaScript i ładujesz Reacta za pomocą tagu `<script>`, jest on dostępny jako globalna zmienna `React`.
```

### Używanie notacji kropkowej w JSX {#using-dot-notation-for-jsx-type}

> Możesz odnosić się do Reactowych komponentów za pomocą notacji kropki. Jest to wygodne rozwiązanie w sytuacji, gdy z jednego modułu eksportujesz wiele komponentów. Przykładowo, jeżeli `MyComponents.DatePicker` jest komponentem, możesz użyć go bezpośrednio w JSX w następujący sposób:

Reviewer: ```suggestion
Możesz odnosić się do reactowych komponentów za pomocą notacji kropkowej. Jest to wygodne rozwiązanie w sytuacji, gdy z jednego modułu eksportujesz wiele komponentów. Przykładowo, jeżeli `MyComponents.DatePicker` jest komponentem, możesz użyć go bezpośrednio w JSX w następujący sposób:
```

```js{10}

> import React from 'react';

Reviewer: Zamień
```<div>Imagine a {props.color} datepicker here.</div>```
na
```<div>Wyobraź sobie, że jest tutaj kalendarz w kolorze {props.color}.</div>```

}
```

> ### Komponenty zdefiniowane przez użytkownika muszą być pisane z wielkiej litery {#user-defined-components-must-be-capitalized}

Reviewer: ```suggestion
### Komponenty zdefiniowane przez użytkownika muszą być pisane wielką literą {#user-defined-components-must-be-capitalized}
```

> Jeżeli nazwa elementu zaczyna się od małej litery, oznacza to odniesienie do wbudowanego komponentu takiego jak na przykład: `<div>` lub `<span>` i skutkuje przekazaniem ciągu znaków `'div'` lub `'span'` do funkcji `React.createElement`. Nazwy typów, które zaczynają się od wielkiej litery, na przykład: `<Foo />` są kompilowane do wywołania `React.createElement(Foo)` i odnoszą się do komponentów zdefiniowanych lub zaimportowanych do Twoich plików zawierających kod JavaScript.

Reviewer: ```suggestion
Jeżeli nazwa elementu zaczyna się od małej litery, oznacza to odniesienie do wbudowanego komponentu, takiego jak na przykład: `<div>` lub `<span>`, i skutkuje przekazaniem ciągu znaków `'div'` lub `'span'` do funkcji `React.createElement`. Nazwy typów, które zaczynają się od wielkiej litery, na przykład: `<Foo />`, są kompilowane do wywołania `React.createElement(Foo)` i odnoszą się do komponentów zdefiniowanych lub zaimportowanych do twoich plików zawierających kod JavaScript.
```

Reviewer: Zgadzam się, że trzeba usunąć te nadmiarowe słowa, ale dodałbym, oprócz tego, wszystkie moje sugestie.

> Rekomendujemy używanie wielkich liter w nazwach komponentów. Jeżeli Twój komponent ma nazwę rozpoczynającą się od małej litery, przypisz ją do zmiennej, której nazwa zaczyna się od wielkiej litery zanim użyjesz go w JSX.

Reviewer: ```suggestion
Rekomendujemy używanie wielkich liter w nazwach komponentów. Jeżeli twój komponent ma nazwę rozpoczynającą się od małej litery, przypisz ją do zmiennej, której nazwa zaczyna się od wielkiej litery, zanim użyjesz go w JSX.
```

> Na przykład, poniższy kod nie zachowa się tak jak można oczekiwać:

Reviewer: ```suggestion
Na przykład, poniższy kod nie zachowa się tak, jak można by tego oczekiwać:
```

```js{3,4,10,11}
import React from 'react';

> // Źle! To jest komponent i jego nazwa powinna zaczynać się z wielkiej litery:

Reviewer: ```suggestion
// Źle! To jest komponent i jego nazwa powinna zaczynać się wielką literą:
```

function hello(props) {

>   // Poprawne! To użycie <div> jest poprawne, bo jest on poprawnym, wbudowanym tagiem HTML:

Reviewer: ```suggestion
  // Dobrze! To użycie <div> jest poprawne, bo jest on właściwym wbudowanym tagiem HTML:
```

>   return <div>Hello {props.toWhat}</div>;

Reviewer: ```suggestion
  return <div>Witaj, {props.toWhat}</div>;
```

>    return <div>Witaj, {props.toWhat}</div>;

Reviewer: ```suggestion
  return <div>Witaj, {props.toWhat}</div>;
```

}

function HelloWorld() {
  // Źle! React traktuje <hello /> jak element wbudowany HTML, bo jego nazwa nie zaczyna się od wielkiej litery:

>   return <hello toWhat="World" />;

Reviewer: ```suggestion
  return <hello toWhat="Świecie" />;
```

>    return <hello toWhat="Świecie" />;

Reviewer: ```suggestion
  return <hello toWhat="Świecie" />;
```

}
```

> Aby to naprawić zmienimy nazwę z `hello` na `Hello` i użycie na `<Hello />`:

Reviewer: ```suggestion
Aby to naprawić, zmienimy nazwę z `hello` na `Hello`, a w miejscu użycia napiszemy `<Hello />`:
```

```js{3,4,10,11}
import React from 'react';

> // Poprawne! To jest komponent więc powinien mieć nazwę pisaną z wielkiej litery:

Reviewer: ```suggestion
// Dobrze! To jest komponent, więc powinien mieć nazwę pisaną wielką literą:
```

function Hello(props) {

>   // Poprawne! To użycie <div> jest poprawne, bo div jest poprawnym elementem HTML:

Reviewer: ```suggestion
  // Dobrze! To użycie <div> jest poprawne, bo div jest właściwym wbudowanym elementem HTML:
```

>   return <div>Hello {props.toWhat}</div>;

Reviewer: ```suggestion
  return <div>Witaj, {props.toWhat}</div>;
```

}

function HelloWorld() {

>   // Poprawne! React traktuje <Hello /> jak komponent, bo jego nazwa jest pisana z wielkiej litery:

Reviewer: ```suggestion
  // Dobrze! React traktuje <Hello /> jak komponent, bo jego nazwa jest zaczyna się od wielkiej litery:
```

>   // Dobrze! React traktuje <Hello /> jak komponent, bo jego nazwa jest zaczyna się od wielkiej litery:

Reviewer: ```suggestion
  // Dobrze! React traktuje <Hello /> jak komponent, bo jego nazwa zaczyna się od wielkiej litery:
```

>    return <Hello toWhat="Świecie" />;

Reviewer: ```suggestion
  return <Hello toWhat="Świecie" />;
```

};

function Story(props) {
  // Źle! Komponent w JSX nie może być wyrażeniem.
  return <components[props.storyType] story={props.story} />;
}
```

> Aby to naprawić przypisujemy najpierw wyrażenie do zmiennej z nazwą zaczynającą się z wielkiej litery:

Reviewer: Myślę, że za dużo tu "ą" ("nazw*ą* zaczynaj**ą**c**ą** się wielk**ą** liter**ą**") i zostawiłbym moją wersję.

> Aby to naprawić przypisujemy najpierw wyrażenie do zmiennej z nazwą zaczynającą się z wielkiej litery:

Reviewer 2: jadnak powinno być `wielką literą` lub `od wielkiej litery`
https://sjp.pwn.pl/poradnia/haslo/wielka-duza-litera-czy-z-wielkiej-duzej-litery;3927.html

Literały łańcuchów znaków można przekazywać jako wartość atrybutu. Poniższe przykłady są równoważne:

```js
<MyComponent message="hello world" />

<MyComponent message={'hello world'} />
```

Gdy przekazujesz literał łańcucha znaków, jego wartość jest odkodowana z HTML'a. Dlatego poniższe wyrażenia będą miały ten sam wynik:

```js
<MyComponent message="&lt;3" />

<MyComponent message={'<3'} />
```

Zazwyczaj, to zachowanie nie jest istotne i zostało wspomniane tylko dla kompletności dokumentacji.

### Atrybuty domyślnie mają wartość "true" {#props-default-to-true}

Gdy przekażesz atrybut bez jawnego podawania wartości, domyślnie przyjmie on wartość `true`. Poniższe wyrażenia JSX są równoważne:

```js
<MyTextBox autocomplete />

<MyTextBox autocomplete={true} />
```

Nie rekomendujemy jednak przekazywania wartości `true` w jawny sposób, bo może być to pomylone ze [skrótowym zapisem definicji obiektu w ES6](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015) `{foo}`, który jest równoznaczny z `{foo: foo}`, a nie `{foo: true}`. To zachowanie JSX zostało zaimplementowane żeby odwzorować sposób działania HTML.

> ### Atrybuty rozszczepione {#spread-attributes}

Reviewer 2: ```suggestion
### Właściwości rozszczepione (ang. spread) {#spread-attributes}
```

> ### Atrybuty rozszczepione {#spread-attributes}

Reviewer: Dla konsekwencji, wrzuć `spread` pomiędzy gwiazdki: `*spread*`.

Możesz dowiedzieć się więcej na temat [warunkowego renderowania](/docs/conditional-rendering.html) i [pętli](/docs/lists-and-keys.html) na odpowiednich stronach.

### Literały tekstowe {#string-literals}

Literały tekstowe można przekazywać jako wartość właściwości. Poniższe przykłady są równoważne:

```js
<MyComponent message="hello world" />

<MyComponent message={'hello world'} />
```

Gdy przekazujesz literał tekstowy, w jego treści zakodowywany jest HTML. Dlatego poniższe wyrażenia będą miały ten sam wynik:

```js
<MyComponent message="&lt;3" />

<MyComponent message={'<3'} />
```

> Zwykle jednak konsekwencje tego zachowania nie są istotne, jednak zostały wspomniane dla kompletności dokumentacji.

Reviewer: ```suggestion
Zwykle jednak konsekwencje tego zachowania nie są istotne, ale zostały wspomniane dla kompletności dokumentacji.
```
Powtórzone "jednak".

<MyTextBox autocomplete={true} />
```

Nie rekomendujemy jednak przekazywania wartości `true` w jawny sposób, bo może być to pomylone ze [skrótowym zapisem definicji obiektu w ES6](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015) `{foo}`, który jest równoznaczny z `{foo: foo}`, a nie `{foo: true}`. To zachowanie w składni JSX zostało zaimplementowane, żeby odwzorować sposób działania atrybutów HTML.

> ### Właściwości rozszczepione (ang. spread) {#spread-attributes}

Reviewer: ```suggestion
### Właściwości rozszczepione (ang. *spread*) {#spread-attributes}
```

Operator rozszczepienia może być bardzo pomocny, lecz używając go, łatwo jest stracić kontrolę nad przekazywanymi atrybutami. Może się zdarzyć, że przekażesz niepoprawny atrybut HTML lub nadmiarowy atrybut, do komponentu, który wcale go nie potrzebuje. Dlatego rekomendujemy ostrożność w użyciu tego rozwiązania.

## Elementy potomne w JSX {#children-in-jsx}

W JSX, jeżeli wyrażenie posiada zarówno tag otwierający jak i zamykający, to jego elementy potomne są przekazywane jako specjalny atrybut: `props.children`. Istnieje kilka sposobów żeby przekazać atrybut `children`:

### Literały łańcuchów znaków {#string-literals-1}

Możesz umieścić łańcuch znaków pomiędzy tagiem otwierającym, a zamykającym. Wtedy wartość `props.children` będzie równa przekazanemu łańcuchowi znaków. Jest to szczególnie przydatne w użyciu z niektórymi wbudowanymi elementami HTML. Na przykład:

```js
<MyComponent>Hello world!</MyComponent>
```

Powyższy kod jest poprawnym wyrażeniem JSX, a wartość `props.children` w komponencie `MyComponent` będzie równa przekazanemu łańcuchowi znaków `"Hello world!". HTML jest odkodowywany więc możesz pisać w JSX tak jak zwykły HTML:

```html

> <div>This is valid HTML &amp; JSX at the same time.</div>

Reviewer: ```suggestion
<div>To jest poprawny kod HTML &mdash; ale też JSX.</div>
```

```

JSX usuwa białe znaki na początku i końcu linii, a także: puste linie, puste linie przylegające do elementów, puste linie wewnątrz łańcuchów znaków. Białe znaki w łańcuchach znaków są zamieniane na pojedyncze spacje. Z tego powodu wszystkie poniższe przykłady są renderowane w ten sam sposób:

```js

> <div>Hello World</div>

Reviewer: ```suggestion
<div>Witaj, świecie</div>
```

Zamień wszędzie.

Wszystkie pozostałe właściwości są przekazywane poprzez rozszczepiony obiekt `...other`, co powoduje, że komponent jest elastyczny pod względem przyjmowanych właściwości. W tym przykładzie przekazane zostały atrybuty: `onClick` i `children`.

Operator rozszczepienia może być bardzo pomocny, lecz używając go, łatwo jest stracić kontrolę nad przekazywanymi właściwościami. Może się zdarzyć, że przekażesz niepoprawny atrybut HTML lub nadmiarową właściwość do komponentu, który wcale go nie potrzebuje. Dlatego zalecamy ostrożność w używaniu tego rozwiązania.

## Elementy potomne w JSX {#children-in-jsx}

W JSX, jeżeli wyrażenie posiada zarówno tag otwierający, jak i zamykający, to jego elementy potomne są przekazywane jako specjalna właściwość: `props.children`. Istnieje kilka sposobów na przekazanie właściwości `children`:

### Literały tekstowe {#string-literals-1}

Możesz umieścić łańcuch znaków pomiędzy tagiem otwierającym a zamykającym. Wtedy wartość `props.children` będzie równa przekazanemu łańcuchowi znaków. Jest to szczególnie przydatne w użyciu z niektórymi wbudowanymi elementami HTML. Na przykład:

```js

>  <MyComponent>Witaj, świecie!</MyComponent>

Reviewer: ```suggestion
<MyComponent>Witaj, świecie!</MyComponent>
```

</div>
```

### Elementy potomne w JSX {#jsx-children}

> Możesz przekazywać więcej niż jeden element jako potomny w JSX. Jest to użyteczne podczas wyświetlania zagnieżdżonych komponentów:

Reviewer: ```suggestion
W JSX możesz przekazywać więcej niż jeden element jako potomny. Przydaje się to przy wyświetlaniu zagnieżdżonych komponentów:
```

</MyContainer>
```

> Przekazywane elementy mogą być różnego typu, a więc możesz używać łańcuchów znaków wraz z innymi rodzajami elementów potomnych. Jest to kolejne podobieńswo do HTML. Poniższy kod jest zarówno poprawnym JSX jak i HTML:

Reviewer: ```suggestion
Przekazywane elementy mogą być różnego typu, a więc możesz używać łańcuchów znaków wraz z innymi rodzajami elementów potomnych. Jest to kolejne podobieńswo do HTML-a. Poniższy kod jest zarówno poprawnym JSX, jak i HTML:
```

```html

> <div>

Reviewer: `Here is a list:` -> `Oto lista:`
`Item #` -> `Element #`

</div>
```

Komponenty mogą również zwracać tablicę elementów::

```js
render() {
  // Nie ma potrzeby otaczać elementów listy dodatkowym elementem JSX!
  return [
    // Pamiętaj o kluczach :)

>     <li key="A">First item</li>,

Reviewer: ```suggestion
     <li key="A">Pierwszy</li>,
```

>     <li key="B">Second item</li>,

Reviewer: ```suggestion
     <li key="B">Drugi</li>,
```

>     <li key="C">Third item</li>,

Reviewer: ```suggestion
    <li key="C">Trzeci</li>,
```

>      <li key="A">Pierwszy</li>,

Reviewer: ```suggestion
    <li key="A">Pierwszy</li>,
```

>      <li key="B">Drugi</li>,

Reviewer: ```suggestion
    <li key="B">Drugi</li>,
```

  ];
}
```

> ### Wyrażenia JavaScript jako komponenty potomne {#javascript-expressions-as-children}

Reviewer: ```suggestion
### Wyrażenia javascriptowe jako komponenty potomne {#javascript-expressions-as-children}
```

> Możesz przekazać dowolne wyrażenie JavaScript obejmując je `{}`. Poniższe wyrażenia są równoważne:

Reviewer: ```suggestion
Możesz przekazać dowolne wyrażenie javascriptowe, obejmując je klamrami `{}`. Poniższe wyrażenia są równoważne:
```

```js
<MyComponent>foo</MyComponent>

<MyComponent>{'foo'}</MyComponent>
```

To rozwiązanie jest często przydatne przy renderowaniu list. Na przykład, poniższy kod renderuje listę elementów HTML:

```js{2,9}

> function Item(props) {

Reviewer: Zamień
```['finish doc', 'submit pr', 'nag dan to review']```
na:
```['dokończyć dokumentację', 'wystawić PR-a', 'namówić Dana na review']```

}
```

> Wyrażenia JavaScript mogą być również używane obok innych typów danych:

Reviewer: ```suggestion
Wyrażenia javascriptowe można również używać razem z innymi typami danych:
```

```js{2}
function Hello(props) {

>   return <div>Hello {props.addressee}!</div>;

Reviewer: ```suggestion
  return <div>Cześć, {props.addressee}!</div>;
```

}
```

### Funkcje jako komponenty potomne {#functions-as-children}

> Wyrażenia JavaScript przekazywane w JSX są rozwiązywane do łańcuchów znaków, elementu Reacta lub listy tych rzeczy. Jednakże `props.children` działa jak każdy inny atrybut i może przekazać dowolny rodzaj danych, nie tylko takie, które React wie jak wyświetlić. Na przykład, twój komponent może przyjmować funkcję zwrotną jako `props.children`:

Reviewer: ```suggestion
Zwykle wyrażenia javascriptowe przekazywane w JSX są przekształcane na łańcuchy znaków, elementy reactowe lub listy tych rzeczy. Jednakże właściwość `props.children` działa jak każda inna i może przekazać dowolny rodzaj danych, nie tylko takie, które React wie, jak wyświetlić. Na przykład, twój komponent może przyjmować funkcję zwrotną jako `props.children`:
```

```js{4,13}
// Wywołuje potomną funkcję zwrotną numTimes, aby powtórzyć wyświetlenie elementów
function Repeat(props) {
  let items = [];

>   for (let i = 0; i < props.numTimes; i++) {

Reviewer: Zamień
```{(index) => <div key={index}>This is item {index} in the list</div>}```
na
```{(index) => <div key={index}>To jest {index}. element listy</div>}```

}
```

> Elementy potomne przekazane do własnych komponentów mogą mieć dowolny typ jeżeli tylko React będzie w stanie zamienić je na zrozumiały dla siebie typ danych przed wyświetleniem. Nie jest to popularny sposób wykorzystania JSX ale można w ten sposób rozszerzać jego możliwości.

Reviewer: ```suggestion
Elementy potomne przekazane do własnych komponentów mogą mieć dowolny typ, pod warunkiem, że React będzie w stanie zamienić je na zrozumiały dla siebie typ danych przed wyświetleniem. To podejście nie jest zbyt popularne w składni JSX, ale można w ten sposób rozszerzać jej możliwości.
```

<div>{true}</div>
```

> Warunkowe renderowanie elementów przez Reacta może być użyteczne. Na przykład, poniższy kod wyrenderuje komponent `<Header />` tylko, gdy wartość zmiennej `showHeader` jest równa `true`:

Reviewer: ```suggestion
Może się to okazać przydatne przy warunkowym renderowaniu elementów. Na przykład, poniższy kod wyrenderuje komponent `<Header />` tylko, gdy wartość zmiennej `showHeader` jest równa `true`:
```

</div>
```

> Warto pamiętać, o tym, że niektóre [fałszywie prawdziwe wartości (ang. falsy values)](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), takie jak `0` będą renderowane przez Reacta. Na przykład, poniższy kod, nie zachowa się tak jak można by na pierwszy rzut oka pomyśleć, ponieważ gdy atrybut `props.messages` będzie pustą tablicą, wartość `0` zostanie wyświetlona:

Reviewer: ```suggestion
Warto pamiętać o tym, że niektóre [fałszopodobne wartości (ang. *falsy values*)](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), takie jak `0`, będą renderowane przez Reacta. Przykładowo, poniższy kod nie zachowa się tak, jak można by na pierwszy rzut oka pomyśleć, ponieważ gdy atrybut `props.messages` będzie pustą tablicą, wyświetlona zostanie wartość `0`:
```

</div>
```

> Aby to naprawić należy upewnić się, że wyrażenie przed `&&` zawsze jest wartością logiczną:

Reviewer: ```suggestion
Aby to naprawić, należy upewnić się, że wyrażenie przed `&&` zawsze jest wartością logiczną:
```

</div>
```

> Jeżeli chcesz, aby wartość taka jak: `false`, `true`, `null`, lub `undefined` została wyświetlona, w pierwszej kolejności należy [skonwertować ją na łańcuch znaków](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#String_conversion):

Reviewer: ```suggestion
Jeżeli chcesz, aby wartość taka jak: `false`, `true`, `null` lub `undefined` została wyświetlona, w pierwszej kolejności należy [przekonwertować ją na łańcuch znaków](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#String_conversion):
```

```js{2}

> <div>

Reviewer: Zamień:
```My JavaScript variable is {String(myVariable)}.```
na
```Moja javascriptowa zmienna to {String(myVariable)}.```
